### ISSUE TITLE
Realm: -ll:ahandlers option causes memory corruption

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#1043]
When running this reproducer on 2 sapling nodes:

[bug1043.tar.gz](https://github.com/StanfordLegion/legion/files/6255125/bug1043.tar.gz)

I get:

```
free(): invalid pointer
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007f4b02134859 in __GI_abort () at abort.c:79
#2  0x00007f4b0219f3ee in __libc_message (action=action@entry=do_abort, fmt=fmt@entry=0x7f4b022c9285 "%s\n") at ../sysdeps/posix/libc_fatal.c:155
#3  0x00007f4b021a747c in malloc_printerr (str=str@entry=0x7f4b022c74ae "free(): invalid pointer") at malloc.c:5347
#4  0x00007f4b021a8cac in _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:4173
#5  0x00007f4b050b050b in Realm::IncomingMessageManager::handler_thread_loop (this=0x5649ad75e840) at /home/manolis/legion/runtime/realm/activemsg.cc:862
#6  0x00007f4b050b1134 in Realm::Thread::thread_entry_wrapper<Realm::IncomingMessageManager, &Realm::IncomingMessageManager::handler_thread_loop> (obj=0x5649ad75e840) at /home/manolis/legion/runtime/realm/threads.inl:97
#7  0x00007f4b04d5b57c in Realm::KernelThread::pthread_entry (data=0x5649b08a5560) at /home/manolis/legion/runtime/realm/threads.cc:774
#8  0x00007f4b020e8609 in start_thread (arg=<optimized out>) at pthread_create.c:477
#9  0x00007f4b02231293 in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95
```

**Comment by lightsighter** [comment]
Everyone in the group should endeavor to find test cases as minimal as this one. ;)

**Comment by streichler** [comment]
Should be fixed with commit 465b4645.  @manopapad can you confirm when you have a moment?

**Comment by manopapad** [comment]
Confirmed fixed
